cmd/mt-whisper-importer-reader/main.go

```
@@ -248,123 +205,113 @@ func shortAggMethodString(aggMethod whisper.AggregationMethod) string {
 	}
 }
 
-func getMetric(w *whisper.Whisper, file string) (*archive.Metric, error) {
+func getMetrics(w *whisper.Whisper, file, name string) (archive.Metric, error) {
+	var res archive.Metric
 	if len(w.Header.Archives) == 0 {
-		return nil, errors.New(fmt.Sprintf("ERROR: Whisper file contains no archives: %q", file))
+		return res, fmt.Errorf("Whisper file contains no archives: %q", file)
 	}
 
-	archives := make([]archive.Archive, 0, len(w.Header.Archives))
-	var chunkSpan uint32
-	var rowKey string
-	var aggMethodStr string
-	name := getMetricName(file)
+	var archives []archive.Archive
+
+	method := shortAggMethodString(w.Header.Metadata.AggregationMethod)
```

maybe that function should return an error instead of working with `""`